src/cockpit/channel.py

```diff
@@ -184,6 +190,8 @@ def do_channel_data(self, channel: str, data: bytes) -> None:
         # Catch errors and turn them into close messages
         try:
             self.do_data(data)
+            if self._send_acks is not None:
```

Comment:
This isn't going to work.  You can't just send the acks immediately on receiving data, or you'll just keep getting data and the channel will have no way to exercise backpressure.  You need to only send the ack after the channel has accepted the data.  That will involve work in ProtocolChannel and AsyncChannel.  In general, I'd say that `do_data()` should be responsible for making sure the ack is sent.